catalog/core/catalog-core-solr/src/main/java/ddf/catalog/source/solr/SolrFilterDelegate.java

```diff
@@ -847,13 +848,16 @@ private SolrQuery getXPathQuery(final String pattern, final String searchPhrase,
     }
 
     private void updateDistanceSort(String propertyName, Point point) {
-        if (sortBy != null && sortBy.getPropertyName() != null) {
-            String sortByPropertyName = sortBy.getPropertyName()
-                    .getPropertyName();
-            if (Result.DISTANCE.equals(sortByPropertyName)
-                    || propertyName.equals(sortByPropertyName)) {
-                isSortedByDistance = true;
-                sortedDistancePoint = point.getY() + "," + point.getX();
+        if (sortBys != null && sortBys.length != 0) {
+            for (SortBy sortBy : sortBys) {
+                String sortByPropertyName = sortBy.getPropertyName()
```

Comment:
If the first `getPropertyName()` is null, the second `getPropertyName()` could potentially cause an NPE